packages/twenty-docker/twenty/entrypoint.sh

```diff
@@ -1,28 +1,37 @@
 #!/bin/sh
 set -e
 
-# Check if the initialization has already been done and that we enabled automatic migration
-if [ "${DISABLE_DB_MIGRATIONS}" != "true" ] && [ ! -f /app/docker-data/db_status ]; then
-    echo "Running database setup and migrations..."
+setup_and_migrate_db() {
+    if [ "${DISABLE_DB_MIGRATIONS}" = "true" ]; then
+        return
+    fi
 
-    # Creating the database if it doesn't exist
-    PGUSER=$(echo $PG_DATABASE_URL | awk -F '//' '{print $2}' | awk -F ':' '{print $1}')
-    PGPASS=$(echo $PG_DATABASE_URL | awk -F ':' '{print $3}' | awk -F '@' '{print $1}')
-    PGHOST=$(echo $PG_DATABASE_URL | awk -F '@' '{print $2}' | awk -F ':' '{print $1}')
-    PGPORT=$(echo $PG_DATABASE_URL | awk -F ':' '{print $4}' | awk -F '/' '{print $1}')
-    PGDATABASE=$(echo $PG_DATABASE_URL | awk -F ':' '{print $4}' | awk -F '/' '{print $2}')
-    PGPASSWORD=${PGPASS} psql -h ${PGHOST} -p ${PGPORT} -U ${PGUSER} -d postgres -tc "SELECT 1 FROM pg_database WHERE datname = '${PGDATABASE}'" | grep -q 1 || \
-    PGPASSWORD=${PGPASS} psql -h ${PGHOST} -p ${PGPORT} -U ${PGUSER} -d postgres -c "CREATE DATABASE \"${PGDATABASE}\""
+    if [ ! -f /app/docker-data/db_status ]; then
+        echo "Running database setup and migrations..."
+
+        # Creating the database if it doesn't exist
+        PGUSER=$(echo $PG_DATABASE_URL | awk -F '//' '{print $2}' | awk -F ':' '{print $1}')
+        PGPASS=$(echo $PG_DATABASE_URL | awk -F ':' '{print $3}' | awk -F '@' '{print $1}')
+        PGHOST=$(echo $PG_DATABASE_URL | awk -F '@' '{print $2}' | awk -F ':' '{print $1}')
+        PGPORT=$(echo $PG_DATABASE_URL | awk -F ':' '{print $4}' | awk -F '/' '{print $1}')
+        PGDATABASE=$(echo $PG_DATABASE_URL | awk -F ':' '{print $4}' | awk -F '/' '{print $2}')
+        PGPASSWORD=${PGPASS} psql -h ${PGHOST} -p ${PGPORT} -U ${PGUSER} -d postgres -tc "SELECT 1 FROM pg_database WHERE datname = '${PGDATABASE}'" | grep -q 1 ||
+            PGPASSWORD=${PGPASS} psql -h ${PGHOST} -p ${PGPORT} -U ${PGUSER} -d postgres -c "CREATE DATABASE \"${PGDATABASE}\""
+
+        # Run setup and migration scripts
+        NODE_OPTIONS="--max-old-space-size=1500" tsx ./scripts/setup-db.ts
+    fi
 
-    # Run setup and migration scripts
-    NODE_OPTIONS="--max-old-space-size=1500" tsx ./scripts/setup-db.ts
     yarn database:migrate:prod
     yarn command:prod upgrade
 
-    # Mark initialization as done
-    echo "Successfuly migrated DB!"
-    touch /app/docker-data/db_status
-fi
+    if [ ! -f /app/docker-data/db_status ]; then
+        # Mark initialization as done
+        echo "Successfully migrated DB!"
```

Comment:
Nit but it feels like:
- "Successfully migrated DB" could be outside of this if, at the same level as database:migrate:prod 
- What this `if` represent is more the fact that the Database was successfully created (maybe just move it to the first if?)
- If we do decide to keep this file, it could be renamed to reflect that it's for the db creation and not really the general db status

Reply:
Okay interesting, I've been refactoring keeping in mind that `db_status` should have been defined after waiting for what could be a long migration/upgrade operations ( for multi-workspaces projects ).

Indeed the name should be quite different, could not find where it's being consumed as a guard. Could you please share assets ? or is it just a good to have for the moment ?